tests/testthat/test-getTableSignature.R

```diff
@@ -111,6 +111,26 @@ for (conn in c(list(NULL), get_test_conns())) {
     })
   }
 
+  if (inherits(conn, "duckdb_connection")) {
+    test_that("getTableSignature() generates signature for update_snapshot() (duckdb_connection)", {
+      expect_identical(
+        getTableSignature(data_update_snapsnot, conn),
+        c(
+          "Date"      = "DATE",
+          "POSIXct"   = "TIMESTAMP",
+          "character" = "STRING",
+          "integer"   = "INTEGER",
+          "numeric"   = "DOUBLE",
+          "logical"   = "BOOLEAN",
+          # ..
+          "checksum"  = "varchar(32)",
```

Comment:
Why is it "varchar(32)" here when "STRING" in 'random_data' (line 228)

Reply:
We only impose our special formats if the three columns are at the end of the input (as in this test case).
Therefore, it is "varchar(32)" here.

In the random_data on line 228, we don't impose our special formats (since the columns are not at the end)
Therfore, it uses the default translation for "character" which for DuckDB is "STRING".